_posts/2021-10-12-devfile-v2-and-the-devworkspace-operator-p1.md

```diff
@@ -0,0 +1,149 @@
+
+This is the first of a series of three articles reviewing the changes introduced with the DevWorkspace. Here we will discuss the changes from the point of view of a Che user. The second part will be about the point of view of a Che administrator. The last part will be dedicated to the DevWorkspace Operator.
+
+## How to enable the DevWorkspace Operator 
+Che default workspace engine is the che-server. The DevWorkspace engine needs to be explicitly enabled. The following [chectl](https://github.com/che-incubator/chectl) command deploys Che on OpenShift configured with the DevWorkspace Operator as workspace engine:
+
+```bash
+chectl server:deploy --installer=olm -p openshift \
+                     --workspace-engine=dev-workspace
+```
+
+The command above works for other Kubernetes distributions too (`-p openshift` should be replaced) but, after chectl has completed, the CheCluster CustomResource requires a patch:
+
+```bash
+# For vanilla Kubernetes only
+kubectl patch checluster/eclipse-che --type=merge -n eclipse-che \ 
+           --patch '{"spec": {"server": { "customCheProperties": {"CHE_INFRA_KUBERNETES_ENABLE__UNSUPPORTED__K8S": "true"}}}}' 
+```
+
+## Changes from a Che user perspective
+
+### New Devfile spec (v2.1)
+
+Here is an example of v2.1 Devfile:
+```yaml
+schemaVersion: 2.1.0
+metadata:
+  name: python-hello-world
+attributes:
+  che-theia.eclipse.org/sidecar-policy: USE_DEV_CONTAINER
+components:
+  - name: python
+    container:
+      image: quay.io/devfile/base-developer-image:ubi8-7bd4fe3
+      volumeMounts:
+        - name: venv
+          path: /home/user/.venv
+      memoryLimit: 512Mi
+      mountSources: true
+  - name: venv
+    volume:
+      size: 1G
+```
+
+The Devfile specification has gone through the release of v2. Here are a few notable changes:
+- It is compatible with the specification of a Kubernetes API. The DevWorkspace CRD is an extension of the Kubernetes API and it’s generated from the Devfile specification.
+- It removes [chePlugin and cheEditor component types](https://github.com/eclipse/che/issues/18669).
+- It introduces the volume component type.
+- Events and parent are two new components types.
+- Besides Che it’s used by the OpenShift Developer Console, `odo` and the [Devfile Docker plugin](https://github.com/devfile/devfile-docker-plugin). 
+
+The documentation for Devfile v2.1 is [https://devfile.io/docs/devfile/2.1.0/](https://devfile.io/docs/devfile/2.1.0/user-guide/index.html). A [migration guide from v1 to v2 of the specification](https://devfile.io/docs/devfile/2.1.0/user-guide/migrating-to-devfile-v2.html) is included. 
+
+### A new way to specify the editor and its plugins
+As mentioned above, version 2 of the Devfile, doesn’t include cheEditor and chePlugins component types anymore. 
+
```

Comment:
maybe reference https://che.eclipseprojects.io/2021/06/23/@florent.benoit-devfile-v2-and-ide-plug-ins.html

Reply:
Thanks, that's a good point. I have added the reference.